plenum/server/replica.py

```diff
@@ -534,14 +534,14 @@ def get_lowest_probable_prepared_certificate_in_view(
                 return n
         return None
 
-    def _setup_for_non_master(self):
+    def _setup_for_non_master(self, is_catchup=True):
         """
         Since last ordered view_no and pp_seq_no are only communicated for
         master instance, `last_ordered_3pc` if backup instance and clear
         last view messages
         :return:
         """
-        if not self.isMaster:
+        if not self.isMaster and (is_catchup or self.last_ordered_3pc[1] == 0):
```

Comment:
It would be great to have a test, that we do not re-calculate it if it's already correct.

Reply:
test_setup_for_non_master() should test it.